python/pyspark/sql/session.py

```diff
@@ -1630,6 +1630,13 @@ def sql(
         -------
         :class:`DataFrame`
 
+        Notes
+        -----
+        In Spark Classic, a temporary view referenced in `spark.sql` is resolved immediately,
+        while in Spark Connect it is lazily evaluated.
```

Comment:
I think this note might be very confusing to users, as data frames in Spark are all lazily evaluated, right? Maybe we can say "it is lazily analyzed". 

We should probably document this as a behavior change for Spark Connect. I am pretty sure there are other behavior changes. Also does this lazy analysis apply to persistent tables and views as well?

Reply:
sounds good, let me update with `it is lazily analyzed`.

Besides temp views, this lazy analysis apply to temp functions / configurations / persistent tables.

If the functions/configurations/tables are changed after `spark.table/sql`, the results may be different from Spark Classic.

Reply:
other dataframe APIs may also have the same behavior change